### PR TITLE
backend/feat/#4826 : enabled latest location fetching for booking and ride entities

### PR DESCRIPTION
**Pull request by JanhaviSoni**
## Type of Change

- [x] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/LocationMapping.hs [hunk 1]
```diff
@@ -32,8 +32,13 @@ create = createWithKV
 countOrders :: (MonadFlow m, EsqDBFlow m r, CacheFlow m r) => Text -> m Int
 countOrders entityId = findAllWithKVAndConditionalDB [Se.Is BeamLM.entityId $ Se.Eq entityId] <&> length
 
-findByEntityId :: (MonadFlow m, EsqDBFlow m r, CacheFlow m r) => Text -> m [LocationMapping] --TODO : SORT BY ORDER
-findByEntityId entityId = findAllWithKVAndConditionalDB [Se.Is BeamLM.entityId $ Se.Eq entityId]
+findByEntityId :: (MonadFlow m, EsqDBFlow m r, CacheFlow m r) => Text -> m [LocationMapping]
+findByEntityId entityId =
+  findAllWithOptionsKV
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/LocationMapping.hs:37]: Why did you change the function from `findAllWithKVAndConditionalDB` to `findAllWithOptionsKV`?

File: Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/LocationMapping.hs [hunk 2]
```diff
@@ -32,8 +32,13 @@ create = createWithKV
 countOrders :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => Text -> m Int
 countOrders entityId = findAllWithKVAndConditionalDB [Se.Is BeamLM.entityId $ Se.Eq entityId] <&> length
 
-findByEntityId :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => Text -> m [LocationMapping] --TODO : SORT BY ORDER
-findByEntityId entityId = findAllWithKVAndConditionalDB [Se.Is BeamLM.entityId $ Se.Eq entityId]
+findByEntityId :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => Text -> m [LocationMapping]
+findByEntityId entityId =
+  findAllWithOptionsKV
```
Comment (hkmangla) [inline Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/LocationMapping.hs:37]: Same here.

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/LocationMapping.hs (hunk 1 above)
Comment (JanhaviSoni) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/LocationMapping.hs:37]: cz `findAllWithOptionsKV` gives flexibility to use pass order by in it's arguments

Comment (hkmangla) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/LocationMapping.hs:37]: There must be some alternative for conditionalDB too. Check with @vijaygupta18

**Review by hkmangla — CHANGES_REQUESTED** [review]

**Review by hkmangla — APPROVED** [review]
